src/materialized/http.rs

```diff
@@ -46,28 +47,57 @@ pub fn match_handshake(buf: &[u8]) -> bool {
     METHODS.contains(&buf)
 }
 
-pub async fn handle_connection<A: 'static + AsyncRead + AsyncWrite + Unpin>(
-    a: A,
-    cmd_tx: UnboundedSender<coord::Command>,
+pub struct Server {
+    cmd_tx: Weak<UnboundedSender<coord::Command>>,
```

Comment:
the idea here is that if the tx is dropped we should just drop the server? I feel like Weak is rare enough that documenting behavior (probably not in a doc comment?) on failure is worth doing if it's not trivially obvious.

Reply:
Indeed, that's the intent, though I just spotted a bug with this approach. I'm going to send you a different PR instead.